### PR TITLE
Refactor supernode to allow for embedding and add more embedding examples

### PR DESCRIPTION
**Pull request by pojntfx**
See #195 

This PR allows for the embedding of the supernode in another application (in my specific case, to write [Go bindings for n2n](https://github.com/pojntfx/gon2n)). In addition to that, it also adds more examples for embedding, which were created in order to show a more advanced custom configuration.

The API was changed to allow for the following "high-level start functions" to be created in the bindings:

[Supernode starter](https://github.com/pojntfx/gon2n/blob/master/pkg/supernode.c)

```c
#include "n2n/n2n.h"

int supernode_configure(n2n_sn_t *sss_node, int lport);

int supernode_open_lport_socket(n2n_sn_t *sss_node);

int supernode_open_mgmt_socket(n2n_sn_t *sss_node, int mgmt_port);

int supernode_start(n2n_sn_t *sss_node, int *keep_running);

int supernode_stop(n2n_sn_t *sss_node);
```

This was achieved by decomposing `sn.c` into `sn_utils.c`, `sn.c` and defining the new public functions in `n2n.h`.

The relevant new examples are [example_edge_embed.c](https://github.com/pojntfx/n2n/blob/dev/example_edge_embed.c) and [example_sn_embed.c](https://github.com/pojntfx/n2n/blob/dev/example_sn_embed.c); I have moved the edge embed example with the `quick_edge_init` function to [example_edge_embed_quick_edge_init.c](https://github.com/pojntfx/n2n/blob/dev/example_edge_embed_quick_edge_init.c).

Please take all of this with a grain of salt. I have never written any C before and followed only a simple C tutorial beforehand (I only know Go/JS/Ruby) ;)

### CODE REVIEW

File: sn_utils.c [hunk 1]
```diff
@@ -0,0 +1,784 @@
...
+    sss->lport = N2N_SN_LPORT_DEFAULT;
+    sss->sock = -1;
+    sss->mgmt_sock = -1;
+
+    return 0; /* OK */
+}
+
+/** Deinitialise the supernode structure and deallocate any memory owned by
+ *  it. */
+extern void sn_term(n2n_sn_t *sss)
```
Comment (emanuele-f) [inline sn_utils.c]: Avoid using `extern` as it is implicit in C for function names. Moreover here the function should be `static` and not `extern` as it is only used within this file. Please fix this for all the functions.

**Review by emanuele-f — CHANGES_REQUESTED** [review]
The file sn.c has too many changes and it's very difficult to review. Please make minimal changes (e.g. avoid changing the existing code indentation and when you move code from sn.c to sn_utils.c put that change into a separate commit) in order to make it possible for a quick review of the changes. Please start with a clean pull request

**Comment by pojntfx** [comment]
Oh, yes it looks like I've kept VSCode's formatter on; I'll go ahead and do that once again, but this time without the formatter and minimal changes. Thanks for clearing up the use of `extern` as well.

**Comment by pojntfx** [comment]
Alright, first of all sorry for taking so long, I was busy with some other development. I'll start a clean PR today.

File: sn_utils.c (hunk 1 above)
Comment (pojntfx) [reply sn_utils.c]: Addressing this in the new PR.

**Pull request closed without merging**